### PR TITLE
#165412938 super admin crud

### PR DESCRIPTION
**Pull request by sengayire**
#### What does this PR do?
- Allow admin to register a user
-  create CRUD endpoints for admin
#### Description of Task to be completed?
Create the following endpoints:
`` POST /api/v1/users/ to register new user ``
`` DELETE /api/v1/users/:id to delete a user ``
`` UPDATE /api/v1/users/:id to update a user information ``
`` GET /api/v1/users/   to fetch all users ``
`` GET /api/v1/users/:id   to fetcha single user ``

#### How should this be manually tested?
clone the repo locally and user postman to test endpoints

#### Any background context you want to provide?
- test endpoints with Mocha and Chai
#### What are the relevant pivotal tracker stories?
- 165412938

### CODE REVIEW

File: src/controllers/AuthLocalController.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import * as validate from '../helpers/validation';
+import status from '../config/status';
+
+/**
+ * A class to handle user local authentication
+ */
+export default class AuthLocalController {
+  /**
+   * @param {object} req
+   * @param {object} res
```
Comment (djallas) [inline src/controllers/AuthLocalController.js]: Use descriptive JSDoc comment

File: src/controllers/AuthLocalController.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        });
+      }
+    } catch (error) {
+      return res.status(status.SERVER_ERROR).send({ error });
+    }
+  }
+
+  /**
+   * @description - login user function
+   * @param {object} req
```
Comment (djallas) [inline src/controllers/AuthLocalController.js]: Provide descriptive JSDoc comment

File: src/controllers/AuthLocalController.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        .status(status.UNAUTHORIZED)
+        .send({ error: `User with ${email} email doesn't exist!!!` });
+    } catch (error) {
+      return res.status(500).send({ error });
+    }
+  }
+
+  /**
+   * @description - methode to update user
+   * @param {object} req
```
Comment (djallas) [inline src/controllers/AuthLocalController.js]: Provide descriptive JSDoc comment

File: src/controllers/AuthLocalController.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+      }
+      return res.status(404).send({ error: 'user not found' });
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * @description function to delete user
+   * @param {object} req
```
Comment (djallas) [inline src/controllers/AuthLocalController.js]: Provide descriptive JSDoc comment

File: src/controllers/AuthLocalController.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+          error: 'Unauthorized access'
+        });
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * @description function for admin to create user
+   * @param {object} req
```
Comment (djallas) [inline src/controllers/AuthLocalController.js]: It's better you go through your codes and descriptive JSDoc comment where applied

File: src/tests/controllers/users.test.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    password: 'pasword',
+    bio: 'my bio goes here',
+    image: null
+  };
+  const badEmail = {
+    email: 'sengayireprinceo133@gmail.com',
+    password: 'password23'
+  };
+  describe('user signup', () => {
+    // it('admin should register new user', (done) => {
```
Comment (djallas) [inline src/tests/controllers/users.test.js]: I think it not necessary to push comments in production

File: src/tests/helpers/passportLocal.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+let user = {};
+chai.use(chaiHttp);
+const newUser = {
+  firstName: 'senga',
+  lastName: 'mugabo',
+  username: 'daprince',
+  email: 'daprince@gmail.com'
+};
+
+describe('test locol passport authenticate', () => {
```
Comment (djallas) [inline src/tests/helpers/passportLocal.js]: This is a typo error

File: src/tests/controllers/users.test.js [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    //     .post('/api/v1/auth/create')
+    //     .send(user)
+    //     .end((err, res) => {
+    //       res.body.should.be.an('object');
+    //       res.status.should.be.equal(status.EXIST);
+    //       res.body.should.have.property('error').equal('Sorry, this account already exists');
+    //       done();
+    //     });
+    // });
+  });
```
Comment (djallas) [inline src/tests/controllers/users.test.js]: Remove unnecessary comments

File: src/controllers/AuthLocalController.js (hunk 1 above)
Comment (sengayire) [reply src/controllers/AuthLocalController.js]: chenge applied

File: src/controllers/AuthLocalController.js (hunk 5 above)
Comment (sengayire) [reply src/controllers/AuthLocalController.js]: yes, change applied

File: src/tests/helpers/passportLocal.js (hunk 7 above)
Comment (sengayire) [reply src/tests/helpers/passportLocal.js]: this was the mistake, but I corrected it

File: src/tests/controllers/users.test.js (hunk 8 above)
Comment (sengayire) [reply src/tests/controllers/users.test.js]: i removed them

**Pull request closed without merging**